tests/network/primary_pod_network.go

```diff
@@ -104,10 +104,15 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 					cleanupVMI(virtClient, vmi)
 				})
 
-				It("should report PodIP/s as its own on interface status", func() {
+				It("should report PodIP/s IPv4 as its own on interface status", func() {
 					vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
 					Eventually(vmiIP).Should(WithTransform(removeCIDR, Equal(vmiPod.Status.PodIP)), "should contain VMI Status IP as Pod status ip")
-					Eventually(vmiIPs).Should(ContainElements(vmi.Status.Interfaces[0].IP, libnet.DefaultIPv6Address), "should contain at IPs the IP and the cloud-init configured ipv6")
+					Eventually(vmiIPs).Should(WithTransform(removeCIDR, ContainElement(vmiPod.Status.PodIP)), "should contain IPv4 reported by guest agent")
```

Comment:
We can check that the vmiIP is vmiIPs[0]. Our API describes that the IP is always the first of IPs

Reply:
This is the race we are trying to fix here, there is moment before guest agenst kicks in that vmiIP has not the CIDR and since this is not going to be refreshed by the Eventually the test fails.

Reply:
But we set both IP and IPs from the same code/place, don't we? And you added the new check at the end of the test, so there is no other check happening after it. I don't see how it fixes a race.

Reply:
yes but we are not refreshing both at  the same time if we do `Eventually(vmiIPs).Should(...vmi.Status...)` the `Should` is interpreted only once so later calls to vmiIPs is not going to affect `Should`